Error injecting rest client caused by PatternSyntaxException when http.nonProxyHosts is set

## Describe the bug

when setting `http.nonProxyHosts` containing a pattern starting with `*` Quarkus fails to inject an instance of a rest client.

Despite a pattern like `*.foo.com` is not a valid regexp, it's a valid value for `nonProxyHosts` according to [Oracle Docs](https://docs.oracle.com/javase/8/docs/technotes/guides/net/proxies.html).

### Expected behavior

Create and inject an instance of a rest client without error

### Actual behavior
When Quarkus needs to inject some rest client it fails if `http.nonProxyHost` is set with a value like `*.foo.com|localhost`
```
org.jboss.resteasy.spi.UnhandledException: java.lang.RuntimeException: Error injecting org.acme.rest.client.FooRestClient org.acme.rest.client.CountriesResource.fooApi
        at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:381)
        at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:218)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:519)
        at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:261)
        at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:161)
        at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
        at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:164)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:247)
        at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
        at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:138)
        at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:93)
        at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:481)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2442)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1476)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.RuntimeException: Error injecting org.acme.rest.client.FooRestClient org.acme.rest.client.CountriesResource.fooApi
        at org.acme.rest.client.CountriesResource_Bean.create(CountriesResource_Bean.zig:189)
        at org.acme.rest.client.CountriesResource_Bean.create(CountriesResource_Bean.zig:212)
        at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:96)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:29)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:26)
        at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:26)
        at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
        at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:26)
        at org.acme.rest.client.CountriesResource_Bean.get(CountriesResource_Bean.zig:244)
        at org.acme.rest.client.CountriesResource_Bean.get(CountriesResource_Bean.zig:260)
        at io.quarkus.arc.impl.ArcContainerImpl.beanInstanceHandle(ArcContainerImpl.java:430)
        at io.quarkus.arc.impl.ArcContainerImpl.beanInstanceHandle(ArcContainerImpl.java:443)
        at io.quarkus.arc.impl.ArcContainerImpl$1.get(ArcContainerImpl.java:266)
        at io.quarkus.arc.impl.ArcContainerImpl$1.get(ArcContainerImpl.java:263)
        at io.quarkus.resteasy.common.runtime.QuarkusConstructorInjector.construct(QuarkusConstructorInjector.java:52)
        at org.jboss.resteasy.plugins.server.resourcefactory.POJOResourceFactory.createResource(POJOResourceFactory.java:71)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:401)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:69)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:492)
        ... 15 more
Caused by: java.util.regex.PatternSyntaxException: Dangling meta character '*' near index 0
*.foo.com
^
        at java.base/java.util.regex.Pattern.error(Pattern.java:2027)
        at java.base/java.util.regex.Pattern.sequence(Pattern.java:2202)
        at java.base/java.util.regex.Pattern.expr(Pattern.java:2068)
        at java.base/java.util.regex.Pattern.compile(Pattern.java:1782)
        at java.base/java.util.regex.Pattern.<init>(Pattern.java:1428)
        at java.base/java.util.regex.Pattern.compile(Pattern.java:1068)
        at io.quarkus.restclient.runtime.QuarkusRestClientBuilder.build(QuarkusRestClientBuilder.java:265)
        at io.quarkus.restclient.runtime.RestClientBase.create(RestClientBase.java:79)
```

## To Reproduce

Steps to reproduce the behavior:
1.  clone the [rest-client-quickstart](https://github.com/maykoone/rest-client-quickstart)
2. `./mvnw quarkus:dev -Dhttp.proxyHost=foo-proxy.com -Dhttp.nonProxyHosts="*.foo.com|localhost|127.*|[::1]"`
3.  `curl http://localhost:8080/country`

### Configuration
Included in the project

## Environment (please complete the following information):

### Output of `uname -a` or `ver`
`Linux desktop 5.4.72-microsoft-standard-WSL2 #1 SMP Wed Oct 28 23:40:43 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`

### Output of `java -version`
```
openjdk version "11.0.11" 2021-04-20
OpenJDK Runtime Environment AdoptOpenJDK-11.0.11+9 (build 11.0.11+9)
OpenJDK 64-Bit Server VM AdoptOpenJDK-11.0.11+9 (build 11.0.11+9, mixed mode)
```

### GraalVM version (if different from Java)

### Quarkus version or git rev
`2.0.1.Final`

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)
```
Apache Maven 3.8.1 (05c21c65bdfed0f71a2f2ada8b84da59348c4c5d)
Maven home: /opt/maven
Java version: 11.0.11, vendor: AdoptOpenJDK, runtime: /usr/lib/jvm/java-11-openjdk
Default locale: en, platform encoding: UTF-8
OS name: "linux", version: "5.4.72-microsoft-standard-wsl2", arch: "amd64", family: "unix"
```

--- a/extensions/resteasy-classic/rest-client/runtime/src/main/java/io/quarkus/restclient/runtime/QuarkusRestClientBuilder.java
+++ b/extensions/resteasy-classic/rest-client/runtime/src/main/java/io/quarkus/restclient/runtime/QuarkusRestClientBuilder.java
@@ -33,8 +33,6 @@ import java.util.Set;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.TimeUnit;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
 
 import javax.enterprise.inject.spi.BeanManager;
 import javax.enterprise.inject.spi.CDI;
@@ -263,49 +261,27 @@ public class QuarkusRestClientBuilder implements RestClientBuilder {
         T actualClient;
         ResteasyClient client;
 
-        ResteasyClientBuilder resteasyClientBuilder;
-        List<String> noProxyHosts = Arrays.asList(
-                getSystemProperty("http.nonProxyHosts", "localhost|127.*|[::1]").split("\\|"));
+        ResteasyClientBuilder resteasyClientBuilder = builderDelegate;
+
         if (this.proxyHost != null) {
-            resteasyClientBuilder = builderDelegate.defaultProxy(proxyHost, this.proxyPort);
+            resteasyClientBuilder.defaultProxy(proxyHost, this.proxyPort);
         } else {
-            String envProxyHost = getSystemProperty("http.proxyHost", null);
-            boolean isUriMatched = false;
-            if (envProxyHost != null && !noProxyHosts.isEmpty()) {
-                for (String s : noProxyHosts) {
-                    Pattern p = Pattern.compile(s);
-                    Matcher m = p.matcher(baseURI.getHost());
-                    isUriMatched = m.matches();
-                    if (isUriMatched) {
-                        break;
-                    }
-                }
-            }
+            String userProxyHost = Optional.ofNullable(getConfiguration().getProperty(PROPERTY_PROXY_HOST))
+                    .filter(String.class::isInstance).map(String.class::cast).orElse(null);
 
-            if (envProxyHost != null && !isUriMatched) {
-                // Use proxy, if defined in the env variables
-                resteasyClientBuilder = builderDelegate.defaultProxy(envProxyHost,
-                        Integer.parseInt(getSystemProperty("http.proxyPort", "80")));
-            } else {
-                // Search for proxy settings passed in the client builder, if passed and use them if found
-                String userProxyHost = Optional.ofNullable(getConfiguration().getProperty(PROPERTY_PROXY_HOST))
-                        .filter(String.class::isInstance).map(String.class::cast).orElse(null);
+            Integer userProxyPort = Optional.ofNullable(getConfiguration().getProperty(PROPERTY_PROXY_PORT))
+                    .filter(Integer.class::isInstance).map(Integer.class::cast).orElse(null);
 
-                Integer userProxyPort = Optional.ofNullable(getConfiguration().getProperty(PROPERTY_PROXY_PORT))
-                        .filter(Integer.class::isInstance).map(Integer.class::cast).orElse(null);
-
+            // configuration of the builder has precedence over environment variables
+            if (userProxyHost != null && userProxyPort != null) {
                 String userProxyScheme = Optional.ofNullable(getConfiguration().getProperty(PROPERTY_PROXY_SCHEME))
                         .filter(String.class::isInstance).map(String.class::cast).orElse(null);
 
-                if (userProxyHost != null && userProxyPort != null) {
-                    resteasyClientBuilder = builderDelegate.defaultProxy(userProxyHost, userProxyPort, userProxyScheme);
-                } else {
-                    // ProxySelector if applicable
-                    selectHttpProxy().ifPresent(
-                            proxyAddress -> builderDelegate.defaultProxy(proxyAddress.getHostString(), proxyAddress.getPort()));
-
-                    resteasyClientBuilder = builderDelegate;
-                }
+                resteasyClientBuilder.defaultProxy(userProxyHost, userProxyPort, userProxyScheme);
+            } else {
+                selectHttpProxy().ifPresent(
+                        proxyAddress -> resteasyClientBuilder.defaultProxy(proxyAddress.getHostString(),
+                                proxyAddress.getPort()));
             }
         }
 

